### DISCUSSION TITLE
Impact of revoking an access token on the refresh token

### DISCUSSION THREAD

**Discussion by sagarshah1983** [ory/hydra#2676] [Q&A] [answered]
Hello everyone, 
I have a question related to oauth2/revoke endpoint - during revoke, does anyone know if it should revoke refresh token as well when revoking bearer access token. I could not find it clear from documentation, but that’s how it was functioning when I tried it.

Appreciate your views.

Regards,
Sagar

**Comment by Benehiko** [comment, marked as answer]
Hi @sagarshah1983

It seems to be the case. Please see the [fosite RevokeToken function](https://github.com/ory/fosite/blob/620d4c148307f7be7b2674fe420141b33aef6075/handler/oauth2/revocation.go#L42-L80). 

Also as the specs note

> If the token passed to the request
   is an access token, the server MAY revoke the respective refresh
   token as well.

https://datatracker.ietf.org/doc/html/rfc7009#section-2.1

**Reply by sagarshah1983** [reply]
Thanks @Benehiko for clarifying and confirming!
